lib/Doctrine/ODM/PHPCR/Mapping/ClassMetadata.php

```diff
@@ -441,6 +441,18 @@ public function validateIdentifier()
     }
 
     /**
+     * Validate whether this class needs to be referenceable.
+     *
+     * @throws MappingException if there is an invalid reference mapping
+     */
+    public function validateReferenceable()
+    {
+        if ($this->uuidFieldName && !$this->referenceable && !'full' === $this->versionable) {
```

Comment:
simple versionable does not imply referenceable. not sure if this check is too restrictive as custom mixins could extend referenceable as well - but worst case you just need to explicitly set referenceable to true in that case.